cmd/cartesi-rollups-node/chainid.go

```diff
@@ -0,0 +1,73 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/cartesi/rollups-node/internal/config"
+)
+
+// Checks if the chain id from the configuration matches the chain id reported
+// by the Ethereum node. If they don't, it returns an error.
+func validateChainId(chainId uint, ethereumNodeAddr string) error {
+	remoteChainId, err := getChainId(ethereumNodeAddr)
+	if err != nil {
+		config.ErrorLogger.Printf("Couldn't validate chainId: %v\n", err)
+	} else if chainId != remoteChainId {
+		return fmt.Errorf(
+			"chainId mismatch. Expected %v but Ethereum node returned %v",
+			chainId,
+			remoteChainId,
+		)
+	}
+	return nil
+}
+
+func getChainId(ethereumNodeAddr string) (uint, error) {
+	// create and send the request
```

Comment:
Why not use go-ethereum to do the call?

Reply:
There's no reason at all, I just forgot that was an option 🤦🏽‍♂️